dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/TrackedEntityAttributeController.java

```diff
@@ -223,4 +225,29 @@ private Map<String, String> getRequiredValues( TrackedEntityAttribute attr, Map<
         return result;
     }
 
+    // --------------------------------------------------------------------------
+    // Hooks
+    // --------------------------------------------------------------------------
+    @Override
+    protected void forceFiltering( final WebOptions webOptions, final List<String> filters )
+    {
+        if ( webOptions == null || !webOptions.isTrue( "indexableOnly" ) )
+        {
+            return;
+        }
+
+        if ( filters.stream().anyMatch( f -> f.startsWith( "id:" ) ) )
+        {
+            throw new IllegalArgumentException(
+                "indexableOnly parameter cannot be set if a separate filter for id is specified" );
+        }
+
+        StringBuilder sb = new StringBuilder();
+        sb.append( "id:in:" );
```

Comment:
you should be able to pass that string into the constructor https://docs.oracle.com/javase/7/docs/api/java/lang/StringBuilder.html#StringBuilder(java.lang.String)